src/NuGet.Clients/VsExtension/VsEvents/OnBuildPackageRestorer.cs

```diff
@@ -490,6 +558,7 @@ var waitDialogFactory
                     return;
                 }
 
+                _packageCount += packages.Count;
```

Comment:
`Presumable we need instance state here because these methods are void`
I'm not sure what exactly you mean by this? But ideally i would have also avoided updating instance variable like this but then it's not really updating instance state which could impact outside. Since _packageCount is a private variable n also being updated in private api